content/FIDO2/FIDO2_WebAuthn_Developer_Guide/WebAuthn_Client_Authentication.adoc

```diff
@@ -0,0 +1,116 @@
+
+**userVerification:** The default value is “preferred”. The “required” value means the operation will fail if the response does not have the UV flag set. The “discouraged” value indicates the RP does not want user verification employed during the operation.
+
+**2.** The JavaScript client calls navigator.credentials.get(). The browser validates the rpId against the origin, hashes the clientData, and calls authenticatorGetAssertion method.
+
+**3.** The authenticator finds a credential that matches the Relying Party ID and prompts the user to consent to the authentication. Assuming the steps are successful, the authenticator creates a new assertion by signing over the clientDataHash and authenticatorData with the private key generated for this account during registration.
+
+**4.** The authenticator returns the authenticatorData and assertion signature back to the browser. 
+
+**5.** The browser resolves the Promise to a PublicKeyCredential that contains the AuthenticatorAssertionResponse which is returned to the RP to finalize the authentication
+
+==== PublicKeyCredential Example:
+[source,javascript]
+----
+{
+  "id": "X9FrwMfmzj...",
+  "response": {
+    "authenticatorData": "xGzvgq0bVGR3WR0Aiwh1nsPm0uy085R0v-ppaZJdA7cBAAAACA",
+    "clientDataJSON": "eyJjaGFsbG...",
+    "signature": "MEUCIQDNrG..."
+  },
+  "clientExtensionResults": {}
+}
+----
+
+**id:** The credential identifier.
+
+**response:** contains the metadata the RP needs to validate the assertion. The **authenticatorData** contains the authenticator data. The **clientDataJSON** contains the JSON-serialized data passed to the authenticator by the client in order to generate the credential. The signature contains the raw signature returned from the authenticator.
+
+**clientExtensionResults():** method returns values for zero or more WebAuthn extensions.
+
+==== Parsed authenticatorData Example:
+[source,javascript]
+----
+{
+    "authData": {
+      "flags": {
+        "AT": false,
+        "ED": false,
+        "UP": true,
+        "UV": false
+      },
+      "rpIdHash": "xGzvgq0bVGR3WR0Aiwh1nsPm0uy085R0v-ppaZJdA7c=",
+      "signatureCounter": 8
+}
+----
+
+**authData:** The authenticator data is a byte array containing data about the get credential operation, including a signature counter and a hash of the RP ID.
+
+**flags:** The **AT** indicates whether the authenticator added attested credential data. The **ED** flag indicates if the authenticator data has extensions. The **UP** flag indicates if the user is present. The **UV** flag indicates if the user is verified (PIN or Biometric).
+
+**rpIdHash:** a SHA-256 has of the RP ID the credential is scoped to.
+
+**signatureCounter:** is incremented for each successful authenticatorGetAssertion operation. It is used by RPs to aid in detecting cloned authenticators.
+
+==== Parsed clientDataJSON Example:
+[source,javascript]
+----
+{
+  "challenge": "kYhXBWX0HO5GstIS02yPJVhiZ0jZLH7PpC4tzJI-ZcA",
```

Comment:
Good to know.  Since the fields are a mix of types, I want to find a better way to convey this to the reader. I will think about it and we can improve it in the next iteration.  How does that sound?

Reply:
Yeah, sounds good.